flutter/news/lib/features/frontpage/data/repositories/articles_repository.dart

```diff
@@ -18,9 +19,10 @@ class ArticlesRepository {
 
   Future<Result<List<Article>>> topHeadlines() async {
     final result = await remoteDataSource.topHeadLines();
-    if (result.isSuccess) {
-      localDataSource.save(topHeadlines: result.data);
-    }
+    result.fold(
+        ifSuccess: (data) => localDataSource.save(topHeadlines: data),
+        ifFailure: (failure) => doNothing(because: "There is nothing to save"));
+
```

Comment:
Maybe I'm missing the value of using Result here. Future already wrap the idea of having a result or an error. 
We should be able to achieve the same without Result

```
// setup
Future<List<String>> remoteDataSource_getTopHeadlines() => Future.value([]);
Future<List<String>> localDataSource_getTopHeadlines() => Future.value([]);
Future<void> localDataSource_save(List<String> headlines) => Future.value(null);

// Implementation
Future<List<String>> getTopHeadlines() async {
  var headlines = await remoteDataSource_getTopHeadlines();
  await localDataSource_save(headlines);
  return localDataSource_getTopHeadlines();
}

// usage
void foo() {
  getTopHeadlines()
      .then((headlines) => debugPrint('headlines: $headlines'))
      .catchError((error) => debugPrint('error: $error'));
}
```

Reply:
Also, do we really need to return the headlines from the localDataSource? Is it not going to return the same headlines that we have just fetched? Is this equivalent?

```Future<List<String>> getTopHeadlines2() async =>
    remoteDataSource_getTopHeadlines().then((headlines) async {
      await localDataSource_save(headlines);
      return headlines;
    });```

Reply:
let's keep this conversation on the usage or not of `Result`. The design of this Repository can be tackled on a different thread

Reply:
This PR is not about introducing Result neither, so please, we can chat about result on other media

Reply:
I had a chat with @gbasile , there are good points on both approaches, `Result` vs `no Result` 

On the `Result` side it unifies error handling for every operation that could result on a failure, not needing to delegate to `Future` for that. Plus extra layer of type safety.

On the `no Result` side, code simplicity and being more aligned with idiomatic constraints.

For now I suggest we keep it, but I might eventually open a PR when we try without Result and see how it looks like